spark/src/main/scala/org/apache/spark/sql/delta/schema/SchemaUtils.scala

```diff
@@ -362,7 +363,12 @@ def normalizeColumnNamesInDataType(
         }
         expression
       }
-      data.select(aliasExpressions: _*)
+      data.queryExecution match {
+        case incrementalExecution: IncrementalExecution =>
```

Comment:
Thanks for fixing this! Could you also add a streaming test to verify the change?

cc @HeartSaVioR to review as well. He found some limitations about `selectFromStreamingDataFrame` in the past. I don't know whether these are fixed or not.